### ISSUE TITLE
[Documentation] External Render

### ISSUE THREAD

**Issue by Th3Whit3Wolf** [go-gitea/gitea#7970]
## Current

```Dockerfile
FROM gitea/gitea:1.6.0
[...]

COPY custom/app.ini /data/gitea/conf/app.ini
[...]

RUN apk --no-cache add asciidoctor freetype freetype-dev gcc g++ libpng python-dev py-pip python3-dev py3-pip
# install any other package you need for your external renderers

RUN pip3 install --upgrade pip
RUN pip3 install -U setuptools
RUN pip3 install jupyter matplotlib docutils 
# add above any other python package you may need to install 
```

This will fail to install jupyter

## Current

```Dockerfile
FROM gitea/gitea:1.9.2
[...]

COPY custom/app.ini /data/gitea/conf/app.ini
[...]

RUN apk --no-cache add asciidoctor freetype freetype-dev gcc g++ libpng python-dev py-pip python3-dev py3-pip py3-zmq
# install any other package you need for your external renderers

RUN pip3 install --upgrade pip
RUN pip3 install -U setuptools
RUN pip3 install jupyter matplotlib docutils 
# add above any other python package you may need to install 
```

Adding the py3-zmq fixes the issue

**Comment by gary-kim** [comment]
Interesting. Want to make a PR?

**Comment by bwenrich** [comment]
Able to reproduce the issue & the suggested solution in Docker 19.03 on Fedora 30, from the gitea/gitea:1.9.2 image. Could not build the image based on gitea/gitea:1.6.0

Before including "py3-zmq", the error is produced (along with much other output from gcc) during the jupyter install
```
Running setup.py install for pyzmq: finished with status 'error'
[...]
fatal error: zmq.h: No such file or directory
  #include "zmq.h"
compilation terminated
[...]
Warning: Couldn't find an acceptable libzmq on the system
```

Please excuse my fork above - I don't mean to be rude or distract from any ongoing PR.
I am studying the Contributing guide & following this Issue to understand the process

**Comment by lunny** [comment]
closed by #8262
